fido/fido.py

```diff
@@ -10,12 +10,14 @@
 from twisted.internet.defer import CancelledError
 from twisted.internet.defer import Deferred
 from twisted.internet.endpoints import TCP4ClientEndpoint
-from twisted.internet.error import TimeoutError as TwistedTimeoutError
+from twisted.internet.error import ConnectError as TwistedConnectError
```

Comment:
Twisted TimeoutError is a sub-class of ConnectError. 
Connection issues can happen for lots of other reasons than just a timeout while establishing the connection (Unkown host, No route to host) so catching ConnectError should take into account a broader set of reasons.